src/modules/accommodations/dto/request/search-accommodation-params.dto.ts

```diff
@@ -0,0 +1,30 @@
+import { IsDate, IsOptional, IsString, Min, IsInt, Max } from 'class-validator';
+
+export class SearchAccommodationQueryParamsDto {
+  @IsString()
+  @IsOptional()
+  search: string;
+
+  @IsString()
+  @IsOptional()
+  category: string;
+
+  @IsDate()
+  @IsOptional()
+  fromDate: Date;
+
+  @IsDate()
```

Comment:
Transform date string into date object if you want to use decorator IsDate. I think you will probably get an error right now if you pass a date as a query param